api/app/src/main/kotlin/packit/controllers/PinController.kt

```diff
@@ -16,4 +23,25 @@ class PinController(
     fun getPinnedPackets(): ResponseEntity<List<PacketMetadata>> {
         return ResponseEntity.ok(pinService.findAllPinnedPackets())
     }
+
+    @PreAuthorize("@authz.hasGlobalPacketManagePermission(#root)")
+    @PostMapping()
+    fun pinPacket(
+        @RequestBody @Validated packetPin: PinDto,
+    ): ResponseEntity<PinDto> {
+        val packetId = packetPin.packetId
+        val pin = pinService.createPinByPacketId(packetId)
+
+        return ResponseEntity(PinDto(pin.packetId), HttpStatus.CREATED)
+    }
+
+    @PreAuthorize("@authz.hasGlobalPacketManagePermission(#root)")
+    @DeleteMapping()
+    fun deletePacket(
```

Comment:
should this be `/{pinId}`.. then follow on and delete by pinid rather than packetId?

Reply:
I did try that at first, but it resulted in overly much overhead of passing pin ids around, so I reverted to using packet ids for all endpoints. The resulting consistency simplified the code.